tools/apigee-x-trial-provision/apigee-x-trial-provision.sh

```diff
@@ -0,0 +1,186 @@
+#!/bin/bash
+# shellcheck disable=SC2059,SC2016
+
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# <http://www.apache.org/licenses/LICENSE-2.0>
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+
+if [ -z "$PROJECT" ]; then
+   echo "ERROR: Environment variable PROJECT is not set."
+   exit 1
+fi
+
+
+# Step 1: Define functions and environment variables
+function token { echo -n "$(gcloud config config-helper --force-auth-refresh | grep access_token | grep -o -E '[^ ]+$')" ; }
+
+PROJECT_NUMBER=$(gcloud projects describe "$PROJECT" --format="value(projectNumber)")
+export PROJECT_NUMBER
+
+export NETWORK=default
+export SUBNET=default
+
+export REGION=${REGION:-europe-west1}
+export ZONE=${ZONE:-europe-west1-b}
+export AX_REGION=${AX_REGION:-europe-west1}
+
+export ORG=$PROJECT
+
+export MIG=apigee-envoy-$REGION
+
+# 
+export RUNTIME_SSL_CERT=~/mig-cert.pem
+export RUNTIME_SSL_KEY=~/mig-key.pem
+export RUNTIME_HOST_ALIAS=$ORG-eval.apigee.net
+
+
+# Step 2: Enable APIs
+gcloud services enable apigee.googleapis.com servicenetworking.googleapis.com compute.googleapis.com cloudkms.googleapis.com --project="$PROJECT"
+
+
+
+# Step 4: Configure service networking
+
+# 1. Define a range of reserved IP addresses for your network. Global
+set +e
+gcloud compute addresses create google-svcs --global --prefix-length=16 --description="Peering range for Google services" --network=default --purpose=VPC_PEERING --project="$PROJECT"
+set -e
+
+# 2. Connect your project's network to the Service Networking API via VPC peering
+gcloud services vpc-peerings connect --service=servicenetworking.googleapis.com --network=default --ranges=google-svcs --project="$PROJECT"
+
+## 4. Create a new eval org using the following command:
+
+gcloud alpha apigee organizations provision \
+  --runtime-location="$REGION" \
+  --analytics-region="$AX_REGION" \
+  --authorized-network=default \
+  --project="$PROJECT"
+
+
+# Step 7: Configure routing, EXTERNAL
```

Comment:
(optional) make EXTERNAL exposure optional

Reply:
raising an issue for this enhancement

Reply:
https://github.com/apigee/devrel/issues/186